### PR TITLE
Feature: create journal to log users actions 

### PR DESCRIPTION
**Pull request by adrkakol**
NEXT STEP:
- [x] add automated cleanup on database

### CODE REVIEW

File: migrations/20220331202635-create-journals.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+              'delete_demand',
+              'delete_all_demands',
+            ],
+          },
+          details: {
+            type: Sequelize.STRING,
+            allowNull: true,
+            defaultValue: null,
+          },
+          user_id: {
```
Comment (michaldomagalski) [inline migrations/20220331202635-create-journals.js:38]: I would suggest not to bind journaling table with user (or any other entity), but instead - save user login in `user` string field. Then, even if the user is completely deleted from database, journal will persist and we can tell that i.e. user `X` has deleted a `place` on April 1, even if user `X` is no longer in db. Binding user with journal by FK will completely erase the history in above scenario.

File: src/demands/demands.controller.ts [hunk 2]
```diff
@@ -121,6 +136,12 @@ export class DemandsController {
         throw new AuthorizationError();
       }
 
+      this.journalsService.logInJournal({
```
Comment (michaldomagalski) [inline src/demands/demands.controller.ts:139]: Inserts into `journals` table should be peformed in the same transaction as actions they log. Right now, every insert has its own transaction, so in this case here - we will end with journaling log stored in db even if deleting demand in line 145 fails.

**Review by michaldomagalski — CHANGES_REQUESTED** [review]

File: src/demands/demands.controller.ts (hunk 2 above)
Comment (michaldomagalski) [reply src/demands/demands.controller.ts:139]: Idea if events is cool, so I am thinking - if we move all `logInJournal` calls in controllers outside the transaction we open, after `COMMIT`, then we are sure that everything is saved properly in db and we can propagate an event.

Example:
```typescript
  public async deleteSomething(id: string): Promise<void> {
    await this.sequelize.transaction(async (transaction) => {
      await this.somethingService.deleteSomething(transaction, id);
    });
    
    this.journalsService.logInJournal({...})
  }
```

File: migrations/20220331202635-create-journals.js (hunk 1 above)
Comment (adrkakol) [reply migrations/20220331202635-create-journals.js:38]: Sure, makes a lot of sense

File: src/demands/demands.controller.ts (hunk 2 above)
Comment (adrkakol) [reply src/demands/demands.controller.ts:139]: Will do as discussed on slack :) Thanks for this

**Review by michaldomagalski — APPROVED** [review]
Nice 💪

**Pull request merged**